src/bootstrap/src/core/builder/mod.rs

```diff
@@ -251,8 +251,10 @@ impl PathSet {
 
     fn has(&self, needle: &Path, module: Kind) -> bool {
         match self {
-            PathSet::Set(set) => set.iter().any(|p| Self::check(p, needle, module)),
-            PathSet::Suite(suite) => Self::check(suite, needle, module),
+            PathSet::Set(set) => {
+                set.iter().any(|p| p.path == needle && Self::check(p, needle, module))
```

Comment:
This feels suspicious. I think we already have a mechanism for skipping things (--exclude on the CLI) so I'd expect that any config.toml mechanism matches the behavior of that. Why do you need to change anything except plumbing through the new config.toml field to match the behavior of `--exclude`?